test/test_datahandling.py

```diff
@@ -162,6 +162,39 @@ def checkFieldConsistancy(self, field):
         for i in range(len(field.axes)):
             self.assertEqual(len(field.axes[i]), field.matrix.shape[i])
 
+    def test_copy(self):
+        c1d = self.f1d.copy(deep=False)
+        assert c1d.matrix is self.f1d.matrix
+
+        assert c1d.axes is not self.f1d.axes
+        assert c1d.axes == self.f1d.axes
+
+        assert c1d.axes_transform_state is not self.f1d.axes_transform_state
+        assert c1d.axes_transform_state == self.f1d.axes_transform_state
+
+        assert c1d.transformed_axes_origins is not self.f1d.transformed_axes_origins
+        assert c1d.transformed_axes_origins == self.f1d.transformed_axes_origins
+
+        assert c1d.infos is not self.f1d.infos
+        assert c1d.infos == self.f1d.infos
+
+        c1d = self.f1d.copy(deep=True)
+        assert c1d.matrix is not self.f1d.matrix
+        np.testing.assert_equal(c1d.matrix, self.f1d.matrix)
+
+        assert c1d.axes is not self.f1d.axes
+        assert c1d.axes == self.f1d.axes
```

Comment:
Kannst du die `assert` statements gegen `self.assertEqual` oder `self.asserTrue` tauschen? Dann kann unittests einen report erstellen:
https://docs.python.org/2/library/unittest.html#basic-example

Reply:
Hmm `unittest` ist ziemlich "basic". `nose` und `pytest` können auch einfache `assert` statements und die funktionen aus `numpy.testing` erkennen. Aber ok, das kann ich für `unittest` umbiegen.

Reply:
ich habe das gerade mal lokal getestet und entgegen meiner Erwartung hat sogar unittest die `assert` statements verstanden.

Ich schlage vor, dass wir uns in Zukunft trotzdem darauf einigen, dass wir keine `assert` statements verwenden, da sie keine Fehlermeldung beinhalten. `numpy.testing.assert*`-Funktionen werfen zwar auch einen `AssertionError`, jedoch mit nuetzlicher Fehlermeldung. Was haelst du davon?